### PR TITLE
NOTAM-6: Added NOTAM class, changed NotamFetcher to make a list of Notam objects, adjusted Notam card attributes to display

### PR DESCRIPTION
**Pull request by Jrubio121**

### CODE REVIEW

**Review by Treeeeeeee — APPROVED** [review]

File: backend/utilities/NotamFetcher.py [hunk 1]
```diff
@@ -26,16 +26,17 @@ def fetch_by_coordinate(self, coordinates, radius = 10):
 
             try:
                 response = requests.get(f"{FAA_API_URL}/notams", headers=self.headers, params=params)
-                print("Status Code:", response.status_code)
```
Comment (jamesf31) [inline backend/utilities/NotamFetcher.py:29]: Do we want to delete this status code? I think it's fine either way.

File: backend/objects/Notam.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        self.number = core_data.get('number', 'N/A')
+        self.purpose = core_data.get('purpose', 'N/A') 
+        self.scope = core_data.get('scope', 'N/A')     
+        self.selection_code = core_data.get('selectionCode', 'N/A')
+        self.series = core_data.get('series', 'N/A')
+        self.traffic = core_data.get('traffic', 'N/A')  
+        self.type = core_data.get('type', 'N/A')
+
+        # Text might not have a default in the original JSON, so consider handling it if needed.
+        self.text = core_data.get('text', 'N/A')
```
Comment (jamesf31) [inline backend/objects/Notam.py:31]: We can decide in the future if we want to use all of this information, but I definitely think that it makes sense to hold onto it for now.

**Review by jamesf31 — APPROVED** [review]

**Review by swagatron300 — APPROVED** [review]

File: backend/objects/Notam.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import json
+
+class Notam:
+    def __init__(self, notam_string):
+        # Parse the input NOTAM JSON string and initialize the instance variables
+        notam_data = json.loads(notam_string)
+        core_data = notam_data['properties']['coreNOTAMData']['notam']
+
+        # Initialize all the specified attributes
+        self.account_id = core_data.get('accountId', 'N/A')
```
Comment (tanminivan) [inline backend/objects/Notam.py:10]: Why "N/A" and not `None`?

File: backend/utilities/NotamFetcher.py [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
+from objects.Notam import Notam  # Import Notam Class
 from objects.Coordinate import Coordinate
 
 class NotamFetcher:
     def __init__(self):
         self.headers = {
-        'client_id': CLIENT_ID,
-        'client_secret': CLIENT_SECRET
+            'client_id': CLIENT_ID,
+            'client_secret': CLIENT_SECRET
```
Comment (tanminivan) [inline backend/utilities/NotamFetcher.py:11]: I am kind of surprised python didn't complain about this before.

File: backend/utilities/NotamFetcher.py [hunk 5]
```diff
@@ -26,16 +26,17 @@ def fetch_by_coordinate(self, coordinates, radius = 10):
...
-                print("Response Text:", response.text)
-
                 response.raise_for_status()
                 data = response.json()
-                current_notam_list = [item['properties']['coreNOTAMData']['notam'] for item in data['items']]
-                notams.extend(current_notam_list)
+
+                # Create Notam objects from the response data
+                for item in data['items']:
+                    notam_string = json.dumps(item)  # Convert the item to a JSON string
```
Comment (tanminivan) [inline backend/utilities/NotamFetcher.py:34]: I'd probably call this `notam_as_json` or something a bit more specific.

File: backend/utilities/NotamFetcher.py [hunk 6]
```diff
@@ -26,16 +26,17 @@ def fetch_by_coordinate(self, coordinates, radius = 10):
...
+                    notam = Notam(notam_string)  # Create a Notam object
+                    notams.append(notam)  # Add to the list
```
Comment (tanminivan) [inline backend/utilities/NotamFetcher.py:36]: can combine into one line: `notams.append( Notam(notam_string) )`

**Review by tanminivan — APPROVED** [review]

File: backend/utilities/NotamFetcher.py (hunk 1 above)
Comment (Jrubio121) [reply backend/utilities/NotamFetcher.py:29]: I think these are useful for debugging while working on it but clutter up the actual app. I can add it back if needed.

**Pull request merged**